src/hacbs/components/Commits/CommitDetailsView.tsx

```diff
@@ -71,19 +92,47 @@ const CommitDetailsView: React.FC<CommitDetailsViewProps> = ({ commitName, appli
     [loaded, pipelineruns],
   );
 
-  const status = React.useMemo(
-    () =>
-      loaded &&
-      Array.isArray(pipelineruns) &&
-      pipelineruns.length > 0 &&
-      pipelineRunFilterReducer(
-        pipelineruns.sort(
-          (a, b) =>
-            parseInt(a.metadata.creationTimestamp, 10) - parseInt(b.metadata.creationTimestamp, 10),
-        )[pipelineruns.length - 1],
-      ),
-    [loaded, pipelineruns],
-  );
+  const sortedPLRList: SortedPLRList = React.useMemo(() => {
+    if (!loaded || loadErr) {
+      return null;
+    }
+
+    const runs = {};
+    pipelineruns.forEach((plr) => {
+      // sort plr into respective status array
+      const plrStatus = pipelineRunFilterReducer(plr);
+      // put task in correct status array
+      statuses.forEach((status) => {
+        if (plrStatus === status) {
+          if (!runs[status]) {
+            runs[status] = [];
+          }
+          runs[status].push(plr);
+        }
+      });
+    });
```

Comment:
you missed to change the suggestion from christian to remove the status map and directly push the `plrStatus` in runs array.